---
title: "WWC_figures"
author: "Erin Eggleston"
date: "10/25/2021"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
options(repos = list(CRAN="http://cran.rstudio.com/"))
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
#load needed packages
#installs are commented out, but add them back as needed
#install.packages("rlang")
packageVersion("rlang") #v1.1.0
#installed.packages("ggplot2")
library(ggplot2); packageVersion("ggplot2") #v3.4.1
theme_set(theme_bw())
library(dplyr); packageVersion("dplyr") #v1.0.8
#install.packages("tidyverse")
library(tidyverse); packageVersion("tidyverse") #v1.1.1
#for decontaminating and filtering
#BiocManager::install("decontam")
library(decontam); packageVersion("decontam") #v1.12.0
#For ordinations and other microbial community analyses
#BiocManager::install("phyloseq")
library(phyloseq); packageVersion("phyloseq")
library(vegan); packageVersion("vegan")
#BiocManager::install("corncob")
library(corncob); packageVersion("corncob"); citation("corncob")
library(reshape2); packageVersion("reshape2")
#install.packages("randomcoloR") 
library(randomcoloR); packageVersion("randomcoloR")
#install.packages("cowplot")
library(cowplot); packageVersion("cowplot")
library(grid); packageVersion("grid") #v 4.1.1
#install.packages("gridGraphics")
library(gridGraphics); packageVersion("gridGraphics") #v 0.5.1
#install.packages("janitor")
library(janitor); packageVersion("janitor") #v 2.2.0
#install.packages("ggpubr")
library(ggpubr); packageVersion("ggpubr") #v 0.5.0
```
```{r}
asv <-read.table("ASVs_countsF.csv", sep = ",", header = TRUE, row.names = 1)
taxa <- as.matrix(read.table("ASVsF_taxonomy.csv", sep = ",", header = TRUE, row.names = 1))
metadata <-read.table("WWC_metadata.csv", sep = ",", header = TRUE, row.names = 1)

asv <- t(asv)

idx<- match(rownames(asv), rownames(metadata))
metadata <-metadata[idx,]
metadata$site
?otu_table
ASV = otu_table(asv, taxa_are_rows = FALSE)
TAX =tax_table(taxa)
META = sample_data(metadata)

ps <- phyloseq(ASV, TAX, META)
str(ps)
# #remove kit.neg and kit.pos
# ps<-subset_samples(ps, site !="kit")
length(get_taxa_unique(ps, "family")) #222 unique family samples
length(get_taxa_unique(ps, "order")) #146 unique order samples
length(get_taxa_unique(ps, "domain")) #3 domains Bacteria, Archean, and NA

get_taxa_unique(ps, "domain") #print unique domains, get rid of mitos, chloros, and NA

ps2 <- ps %>%
  phyloseq::subset_taxa(family !="Mitochondria") %>%
  phyloseq::subset_taxa(order !="Chloroplast") %>%
  phyloseq::subset_taxa(domain !="NA")

length(get_taxa_unique(ps2, "family")) #218
length(get_taxa_unique(ps2, "order")) #118
length(get_taxa_unique(ps2, "domain")) #2

```
```{r}
# save the tables with no Mitochondrial/Chloroplast/Unknown sequences for use in other analyses 
write.table(as(otu_table(ps2), "matrix"),"ASVsF_no_chloromito.csv",sep=",",col.names=NA)
write.table(as(tax_table(ps2), "matrix"),"taxaF_no_chloromito.csv",sep=",",col.names=NA)
write.table(as(sample_data(ps2), "matrix"),"metadata_no_chloromito.csv",sep=",",col.names=NA)
```


```{r}
# Load in the chloro/mito filtered data 
asv <- read.table("ASVsF_no_chloromito.csv",sep=",",header=TRUE, row.names=1)
taxon <- as.matrix(read.table("taxaF_no_chloromito.csv", sep=",", header=TRUE, row.names=1))
meta <- read.table("metadata_no_chloromito.csv",sep=",",header=T,row.names=1)

ASV = otu_table(asv, taxa_are_rows = FALSE)
TAX = tax_table(taxon)
META = sample_data(meta)

ps <- phyloseq(ASV, TAX, META)
str(ps)
```

```{r}
# Filtering using the combined method for DNA extraction-based contaminants
contamdf.com <- isContaminant(ps, method = "combined", neg="is.neg", conc = "quant_reading")
head(contamdf.com)
table(contamdf.com$contaminant) #Looks like the combined method found 1 contaminants.
#Which ASVs were determined to be contaminants?
contamdf.com[contamdf.com$contaminant == "TRUE",]
#one contaminant ASV: ASV_1611
```

```{r}
#Make a prevalence plot using the presence-absence info
# Make data.frame of prevalence in real samples, positive control, and negative samples
# Take a look at how Decontam marked out the different contaminants
# Make phyloseq object of presence-absence in negative controls and true samples
ps.pa <- transform_sample_counts(ps, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$is.neg == "TRUE", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$is.neg == "FALSE", ps.pa)
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                      contaminant=contamdf.com$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) + geom_point() +
  xlab("Prevalence (Negative Controls)") + ylab("Prevalence (True Samples)")

#shows no true contamination in samples
```

```{r}
# Check out some more contaminants using a similar graph as in the frequency method earlier, to make sure all ID'd contaminants "LOOK" like contaminants.

set.seed(100)
plot_frequency(ps, taxa_names(ps)[which(contamdf.com$contaminant)], conc="quant_reading") +
    xlab("DNA Concentration (QuantStudio3 read of library input concentrations ng/ul)")
```



```{r}
# Not all of these "contaminants" were present in DNA extraction controls. Only those in DNA extraction controls, where the frequency is greater than 0 are the following:
contaminants <- c("ASV_1611")

#add an asv column to the contamdf.com dataframe and a column with the contaminants
contamdf.com$asv = rownames(contamdf.com)
contamdf.com$truecontam <- contamdf.com$asv %in% contaminants

#Check that this worked...should be 1 "TRUE"
table(contamdf.com$truecontam)

ps.noncontam <- prune_taxa(!contamdf.com$truecontam, ps)
ps.noncontam #Now we have 1 fewer ASVs because we removed these

write.table(as(otu_table(ps.noncontam), "matrix"),"ASV_decontam.csv", sep = ",", col.names = NA)
write.table(as(tax_table(ps.noncontam),"matrix"), "taxon_decontam.csv", sep = ",", col.names = NA)
write.table(as(sample_data(ps.noncontam), "matrix"),"metadata_decontam.csv", sep = ",", col.names = NA)

#Make a table summarizing the contaminants you removed. 
DNAextraction.contam <- cbind(contamdf.com[contamdf.com$truecontam == TRUE,], tax_table(ps)[rownames(tax_table(ps)) %in% contaminants,])

DNAextraction.contam
write.table(DNAextraction.contam, "Appendix_1_DNAextraction_contaminants_decontam.csv", sep = ",", col.names = NA) #Summary table - part of Appendix
```


```{r}
#Upload data
#import the ASV data that has been filtered for contaminants and low abundance reads. 
ASV <- read.table("ASV_decontam.csv", sep = ",", row.names = 1, header = TRUE)
taxa <- as.matrix(read.table("taxon_decontam.csv", sep = ",", row.names = 1, header = TRUE))
metadata <- read.table("metadata_decontam.csv", sep = ",", header = TRUE, row.names = 1)

meta_bray <-read.table("metadata_decontam.csv", sep = ",", header = TRUE)

asv_dc = otu_table(ASV, taxa_are_rows = FALSE)
taxa_dc = tax_table(taxa)
meta_dc = sample_data(metadata)

ps.dc <- phyloseq(asv_dc,taxa_dc,meta_dc)
```


```{r}
#For Bray-Curtis, turn data into relative abundances since these data are compositional.
relabund <- function(sample) { #Write a relative abundance function
  x = sample/sum(sample)
  x = x*100
  return(x)
}

#pcoa all data bray-curtis, relative abundance
ASV.all<- apply (ASV, 1, relabund)
dim(ASV.all) #see how many ASVs are there
ASV.all <- ASV.all[rowSums(ASV.all) != 0, ] #get rid of ASVs that have no abundance in any samples
dim(ASV.all) #check to see how many ASVs are left. Looks like 3203 ASVs are left.
ASV.all <- t(ASV.all) #make sure sample names are rows
ASV.all.log <- log(ASV.all+1) #Log transform the relative abundance data, where I add a 1 pseudocount to everything

#Adjust the metadata to have only CCC, ice, and control samples of interest

meta_cave1<- meta_bray%>%filter(name %in% c("E1","E2", "E3","E5", "E6","E7","E8","E9","E10","E12","E13","I1","Positive","Negative" ))
  
#Remove the samples from the ASV table that are no longer represented in the metadata tibble
idx <- match(meta_cave1$X, rownames(ASV))
ASV.cave1 <- ASV[idx, ]

ASV.cave <- apply(ASV.cave1, 1, relabund) #Apply the relative abundance function to each row (sample) in the ASV data frame
dim(ASV.cave) #see how many ASVs there are
ASV.cave <- ASV.cave[rowSums(ASV.cave) != 0, ] #get rid of ASVs that have no abundance in any samples
dim(ASV.cave) #check to see how many ASVs are left. Looks like 1808 ASVs are left.
ASV.cave <- t(ASV.cave) #make sure sample names are rows
ASV.cave.log <- log(ASV.cave+1) #Log transform the relative abundance data, where I add a 1 pseudocount to everything



#Helpful reference - http://r-sig-ecology.471788.n2.nabble.com/Variability-explanations-for-the-PCO-axes-as-in-Anderson-and-Willis-2003-td6429547.html
ASV.cave.log.b <- vegdist(ASV.cave.log, "bray") #calculate bray curtis dissimilarity
pcoa.cave <- cmdscale(ASV.cave.log.b, k=(3), eig=TRUE) #metric dimensional scaling (principal coordinates analysis)
barplot(pcoa.cave$eig) #take a look at what principal coordinates best explain the data. There is a small difference between 2 and 3, so also plot principal coordinates 2 and 3.
pcoa.eig <- eigenvals(pcoa.cave) #extract eigenvalues
##show pc explanation fig
Variance <- pcoa.eig / sum(pcoa.eig) #calculate the percent of variance explained by each axis
Variance1 <- 100 * signif(Variance[1], 2) #extract percent explained by variance 1
Variance2 <- 100 * signif(Variance[2], 2) #extract percent explained by variance 2
Variance3 <- 100 * signif(Variance[3], 3) #extract percent explained by variance 3
  
#GGPLOT2 plotting
pcoa.cave2 <- cbind(pcoa.cave$points, meta_cave1) # combine the Pcoa values with metadata for easier plotting
pcoa.cave2$type <- factor(pcoa.cave2$type) #adjust the levels so they plot correctly!

#principal coordinates 1 and 2
ggplot(pcoa.cave2, aes(x = pcoa.cave2[,1]*(-1), y = pcoa.cave2[,2], colour = description, shape = type)) +
  geom_point(size = 4, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1, "%)"), y = paste0("PCoA2 (",Variance2, "%)"), title = "WWC CCC and Ice block PCoA of Bray-Curtis Dissimilarity", shape= "Sample Type", colour = "Sample Description")+
  theme(legend.text.align = 0)
  


#principal coordinates 1 and 3
pcoa.cave2 %>% ggplot(aes(x = pcoa.cave2[,1]*(-1), y = pcoa.cave2[,3], colour = description, shape = type)) +
  geom_point(size = 4, stroke = 2) +
  #coord_fixed() + #aspect ratio of 1
  labs(x = paste0("PCoA1 (", Variance1, "%)"), y = paste0("PCoA3 (",Variance3, "%)"), title = "WWC CCC and Ice block PCoA of Bray-Curtis Dissimilarity", shape= "Sample Type", colour = "Sample Description")+
  theme(legend.text.align = 0)
  
  
#Subset different sample types if needed

# # Subset CCC samples. I will try merging the phyloseq objects later. 
# ps.noncontam.ccc <- ps.noncontam %>%
#   phyloseq::subset_samples(type == "CCC")
# 
# ps.noncontam.water <- ps.noncontam %>%
#   phyloseq::subset_samples(type == "Water")
#   
# ps.noncontam.biofilm <- ps.noncontam %>%
#   phyloseq::subset_samples(type == "Biofilm")
# 
# ps.noncontam.ice <- ps.noncontam %>%
#   phyloseq::subset_samples(type == "Ice") 
# 
# ps.noncontam.background <- ps.noncontam %>%
#   phyloseq::subset_samples(type == "Background")
#
```

```{r}
#Adjust the metadata to have CCC, ice, WATER, and control samples of interest (added WWC water)

meta_cave2<-meta_bray%>%filter(name %in% c("E1","E2","E3","E5", "E6","E7","E8","E9","E10","E12","E13","I1","W2","W3","W5","Positive","Negative"))
  
#Remove the samples from the ASV table that are no longer represented in the metadata tibble
idx2 <- match(meta_cave2$X, rownames(ASV))
ASV.cave2 <- ASV[idx2, ]

ASV.cave2 <- apply(ASV.cave2, 1, relabund) #Apply the relative abundance function to each row (sample) in the ASV data frame
dim(ASV.cave2) #see how many ASVs there are
ASV.cave2 <- ASV.cave2[rowSums(ASV.cave2) != 0, ] #get rid of ASVs that have no abundance in any samples
dim(ASV.cave2) #check to see how many ASVs are left. Looks like 1808 ASVs are left.
ASV.cave2 <- t(ASV.cave2) #make sure sample names are rows
ASV.cave.log2 <- log(ASV.cave2+1) #Log transform the relative abundance data, where I add a 1 pseudocount to everything



#Helpful reference - http://r-sig-ecology.471788.n2.nabble.com/Variability-explanations-for-the-PCO-axes-as-in-Anderson-and-Willis-2003-td6429547.html
ASV.cave.log.b2 <- vegdist(ASV.cave.log2, "bray") #calculate bray curtis dissimilarity
pcoa.cave3 <- cmdscale(ASV.cave.log.b2, k=(3), eig=TRUE) #metric dimensional scaling (principal coordinates analysis)
barplot(pcoa.cave3$eig) #take a look at what principal coordinates best explain the data. WHen including the water samples, principal coordinates 1 and 2 have the most explanatory values. We will plot these.
pcoa.eig2 <- eigenvals(pcoa.cave3) #extract eigenvalues
##show pc explanation fig
Variance_a <- pcoa.eig2 / sum(pcoa.eig2) #calculate the percent of variance explained by each axis
Variance1a <- 100 * signif(Variance_a[1], 2) #extract percent explained by variance 1
Variance2a <- 100 * signif(Variance_a[2], 2) #extract percent explained by variance 2
Variance3a <- 100 * signif(Variance_a[3], 3) #extract percent explained by variance 3
Variance3a  
#GGPLOT2 plotting
pcoa.cave4 <- cbind(pcoa.cave3$points, meta_cave2) # combine the PCoA values with metadata for easier plotting
pcoa.cave4$type <- factor(pcoa.cave4$type) #adjust the levels so they plot correctly!

#principal coordinates 1 and 2

#filled in
ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,2], colour = description, shape = type)) +
  geom_point(size = 4, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA2 (",Variance2a, "%)"), shape= "Sample Type", colour = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(15,17,4,3,16))+
  scale_color_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered2","orange1","lightskyblue1","gray40","mediumblue","yellow2","goldenrod2"))+
  theme(legend.text.align = 0)




#fillable
ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,2], fill = description, shape = type)) +
  geom_point(size = 4, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA2 (",Variance2a, "%)"), shape= "Sample Type", fill = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(22,24,4,3,21))+
  scale_fill_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered2","orange1","lightskyblue1","gray40","mediumblue","yellow2","goldenrod2"))+
  theme(legend.text.align = 0)

#final and saved figure pcoa 1 and pcoa2
ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,2], colour = description, shape = type)) +
  geom_point(size = 3, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA2 (",Variance2a, "%)"), shape= "Sample Type", colour = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(22,24,4,3,21))+
  scale_color_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered3","tomato","lightskyblue1","gray40","mediumblue","darkorange2","goldenrod3"))+
  theme(legend.text.align = 0)  

#final and saved supp figure pcoa 1 and pcoa 3

ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,3], colour = description, shape = type)) +
  geom_point(size = 3, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA3 (",Variance3a, "%)"), shape= "Sample Type", colour = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(22,24,4,3,21))+
  scale_color_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered3","tomato","lightskyblue1","gray40","mediumblue","darkorange2","goldenrod3"))+
  theme(legend.text.align = 0) 

pcoa1_2<-ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,2], colour = description, shape = type)) +
  geom_point(size = 4, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA2 (",Variance2a, "%)"), shape= "Sample Type", colour = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(22,24,4,3,21))+
  scale_color_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered3","tomato","lightskyblue1","gray40","mediumblue","darkorange2","goldenrod3"))+
  theme(legend.text.align = 0)

pcoa1_3 <-ggplot(pcoa.cave4, aes(x = pcoa.cave4[,1]*(-1), y = pcoa.cave4[,3], colour = description, shape = type)) +
  geom_point(size = 3, stroke = 2) +
  ylim(-0.4,0.6)+
  labs(x = paste0("PCoA1 (", Variance1a, "%)"), y = paste0("PCoA3 (",Variance3a, "%)"), shape= "Sample Type", colour = "Sample Description")+
    scale_shape_manual(labels=c("Cave Cryogenic Carbonates","Ice","Negative Control","Positive Control","Water"), values = c(22,24,4,3,21))+
  scale_color_manual(labels= c("Frozen Freeway - cave right", "Frozen Freeway - cave left", "Frozen Freeway Speleothem", "Ice Block","Control","Skating Rink Water", "Skating Rink - cave left", "Skating Rink - cave right"), values = c("orangered","orangered3","tomato","lightskyblue1","gray40","mediumblue","darkorange2","goldenrod3"))+
  theme(legend.text.align = 0) 

png(file="WWC_PCoA_Fig4.png", width=8, height = 5, units='in', res = 300)
plot(pcoa1_2)
dev.off()

png(file="WWC_PCoA_SuppFig.png", width=8, height = 5, units='in', res = 300)
plot(pcoa1_3)
dev.off()

  
```


```{r}
#Stacked bar chart of microbial relative abundances by sample type

# Transform counts to relative abundance to visualize bar charts

ps_ra<-transform_sample_counts(ps.noncontam, function(OTU) OTU/sum(OTU))
ps_ra
str(ps_ra)
sample_data(ps_ra)$type<-factor(sample_data(ps_ra)$type,levels=c("CCC","Ice","kit.negative","kit.positive", "Water"))
sample_data(ps_ra)$description <- factor(sample_data(ps_ra)$description, levels = c("Skating Rink, right", "Skating Rink, left", "Frozen Freeway ,right", "Frozen Freeway, left, top of small speleothem","Frozen Freeway left", "ice block"))

#filter for only taxa >.005 relative abundance across all samples
ps_ra_top <- filter_taxa(ps_ra,function(x) sum (x) > .005, TRUE)


#figure out how many colors you need
length(get_taxa_unique(ps_ra, "order")) #118
length(get_taxa_unique(ps_ra, "class")) #48
length(get_taxa_unique(ps_ra, "phylum")) #26
length(get_taxa_unique(ps_ra, "family")) #218

#how many colors for filtered rel abundance ps object?
length(get_taxa_unique(ps_ra_top, "order")) #58
length(get_taxa_unique(ps_ra_top, "class")) #24
length(get_taxa_unique(ps_ra_top, "phylum")) #17
length(get_taxa_unique(ps_ra_top, "family")) #100


ps_ra_CCC <- subset_samples(ps_ra, type == "CCC")
#add shorthand CCC names
get_variable(ps_ra_CCC, "description")
sample_data(ps_ra_CCC)$short_CCC<-c("FF Right", "FF Left", "SR Right","SR Right","SR Right","SR Right","SR Right","SR Left","SR Left","SR Left", "FF Left")
get_variable(ps_ra_CCC, "short_CCC")
head(sample_data(ps_ra_CCC))

sample_data(ps_ra_CCC)$CCC_short<-c("FF","FF","SR","SR","SR","SR","SR","SR","SR","SR","FF")
sample_data(ps_ra_CCC)$CCC_short<-as.factor(sample_data(ps_ra_CCC)$CCC_short)
get_variable(ps_ra_CCC, "CCC_short")
head(sample_data(ps_ra_CCC))
view(sample_data(ps_ra_CCC))

ps_ra_kitcontrol <- subset_samples(ps_ra, type %in% c("kit.negative", "kit.positive"))
ps_ra_ice <- ps_ra %>% subset_samples(type == "Ice") %>% subset_samples(description=="ice block")
ps_ra_water <- ps_ra %>% subset_samples(type == "Water") %>% subset_samples(site == "WWC")

#you can make c/p/o any number of colors you want; with as much difference between the colors as possible (distinct colors)
c <- 48
palette_c <- distinctColorPalette(c)
p <-26
palette_p <- distinctColorPalette(p)
o <-118
palette_o <- distinctColorPalette(o)
```




```{r}
# Create all sample type graphs from the level of phylum

p1 <- plot_bar(subset_samples(ps_ra_CCC), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p) +
  ggtitle("CCC") +
  # scale_x_discrete(ps_ra_CCC$name)+
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p2 <-  plot_bar(subset_samples(ps_ra_kitcontrol), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p3 <-  plot_bar(subset_samples(ps_ra_ice), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p4 <-  plot_bar(subset_samples(ps_ra_water), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot PHYLUM graph with common legend
bottom_row_P<- plot_grid(p3,p4,p2, labels = c("B","C","D"), ncol = 3, nrow = 1)


P_grid <-plot_grid(p1,bottom_row_P, labels = c("A",""), ncol = 1)

p_legend <- plot_bar(subset_samples(ps_ra_CCC), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  scale_fill_manual(values=palette_p) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legend <- get_legend(
  p_legend +
    theme(legend.position = "bottom")
)
  
plot_grid(P_grid, legend, ncol = 1, rel_heights = c(1, .5))  

png(file="WWC_phylum_barplot.png", width=10, height = 8, units='in', res = 300)
plot_grid(P_grid, legend, ncol = 1, rel_heights = c(1, .5))
dev.off()


```



```{r}
# Create all sample type graphs from the level of class

c1 <- plot_bar(subset_samples(ps_ra_CCC), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c) +
  ggtitle("CCC") +
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c2 <-  plot_bar(subset_samples(ps_ra_kitcontrol), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c3 <-  plot_bar(subset_samples(ps_ra_ice), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c4 <-  plot_bar(subset_samples(ps_ra_water), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot CLASS graph with common legend
bottom_row_C<- plot_grid(c3,c4,c2, labels = c("B","C","D"), ncol = 3, nrow = 1)


C_grid <-plot_grid(c1,bottom_row_P, labels = c("A",""), ncol = 1)

c_legend <- plot_bar(subset_samples(ps_ra_CCC), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  scale_fill_manual(values=palette_c) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legendC <- get_legend(
  c_legend +
    theme(legend.position = "bottom")
)
  
plot_grid(C_grid, legendC, ncol = 1, rel_heights = c(1, .5))  

png(file="WWC_class_barplot.png", width=10, height = 8, units='in', res = 300)
plot_grid(C_grid, legendC, ncol = 1, rel_heights = c(1, .5))
dev.off()
  
```

```{r}
# Create all sample type graphs from the level of order

o1 <- plot_bar(subset_samples(ps_ra_CCC), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o) +
  ggtitle("CCC") +
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o2 <-  plot_bar(subset_samples(ps_ra_kitcontrol), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o3 <-  plot_bar(subset_samples(ps_ra_ice), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o4 <-  plot_bar(subset_samples(ps_ra_water), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot order graph with common legend
bottom_row_O<- plot_grid(o3,o4,o2, labels = c("B","C","D"), ncol = 3, nrow = 1)


O_grid <-plot_grid(o1,bottom_row_O, labels = c("A",""), ncol = 1)

o_legend <- plot_bar(subset_samples(ps_ra_CCC), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  scale_fill_manual(values=palette_o) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legendO <- get_legend(
  o_legend +
    theme(legend.position = "bottom")
)

O_grid
legendO
plot_grid(O_grid, legendO, ncol = 1, nrow = 2, rel_heights = c(100, 0.1))
  
plot_grid(O_grid, legendO, ncol = 1, rel_heights = c(100, 0.1))

ggdraw(plot_grid(plot_grid(o1,bottom_row_O, labels = c("A",""), ncol =1),
                 plot_grid(NULL, legendO, ncol = 2),
                 rel_widths = c(1,0.01)))

```

```{r}
#repeat with ps_ra_top for most abundant taxa
ps_ra_CCC_top <- subset_samples(ps_ra_top, type == "CCC")
#add shorthand CCC names
get_variable(ps_ra_CCC_top, "description")
sample_data(ps_ra_CCC_top)$short_CCC<-c("FF Right", "FF Left", "SR Right","SR Right","SR Right","SR Right","SR Right","SR Left","SR Left","SR Left", "FF Left")
get_variable(ps_ra_CCC_top, "short_CCC")
head(sample_data(ps_ra_CCC_top))

ps_ra_kitcontrol_top <- subset_samples(ps_ra_top, type %in% c("kit.negative", "kit.positive"))
ps_ra_ice_top <- ps_ra_top %>% subset_samples(type == "Ice") %>% subset_samples(description=="ice block")
ps_ra_water_top <- ps_ra_top %>% subset_samples(type == "Water") %>% subset_samples(site == "WWC")

#you can make c/p/o any number of colors you want; with as much difference between the colors as possible (distinct colors)
c_top <- 24
palette_c_top <- distinctColorPalette(c_top)
p_top <-17
palette_p_top <- distinctColorPalette(p)
o_top <-58
palette_o_top <- distinctColorPalette(o)
```

```{r}
# Create all sample type graphs from the level of phylum for most abundant ASvs

p1_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p_top) +
  ggtitle("CCC") +
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p2_top <-  plot_bar(subset_samples(ps_ra_kitcontrol_top), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p_top) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p3_top <-  plot_bar(subset_samples(ps_ra_ice_top), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p_top) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

p4_top <-  plot_bar(subset_samples(ps_ra_water_top), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_p_top) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot PHYLUM graph with common legend
bottom_row_P_top<- plot_grid(p3_top,p4_top,p2_top, labels = c("B","C","D"), ncol = 3, nrow = 1)


P_grid_top <-plot_grid(p1_top,bottom_row_P_top, labels = c("A",""), ncol = 1)

p_legend_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="phylum")+
  geom_bar(aes(fill=phylum), stat="identity",position="stack") +
  scale_fill_manual(values=palette_p_top) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legend_top <- get_legend(
  p_legend_top +
    theme(legend.position = "bottom")
)
  
plot_grid(P_grid_top, legend_top, ncol = 1, rel_heights = c(1, .5))  


```


```{r}
# Create all sample type graphs from the level of class

c1_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c_top) +
  ggtitle("CCC") +
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c2_top <-  plot_bar(subset_samples(ps_ra_kitcontrol_top), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c_top) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c3_top <-  plot_bar(subset_samples(ps_ra_ice_top), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c_top) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

c4_top <-  plot_bar(subset_samples(ps_ra_water_top), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_c_top) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot CLASS graph with common legend
bottom_row_C_top<- plot_grid(c3_top,c4_top,c2_top, labels = c("B","C","D"), ncol = 3, nrow = 1)


C_grid_top <-plot_grid(c1_top,bottom_row_P_top, labels = c("A",""), ncol = 1)

c_legend_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="class")+
  geom_bar(aes(fill=class), stat="identity",position="stack") +
  scale_fill_manual(values=palette_c_top) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legendC_top <- get_legend(
  c_legend_top +
    theme(legend.position = "bottom")
)
  
plot_grid(C_grid_top, legendC_top, ncol = 1, rel_heights = c(1, .5))  
  
```

```{r}
# Create all sample type graphs from the level of order

o1_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o_top) +
  ggtitle("CCC") +
  facet_grid(. ~ short_CCC, scales = "free", space = "free")+
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o2_top <-  plot_bar(subset_samples(ps_ra_kitcontrol_top), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o_top) +
  ggtitle("Controls") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o3_top <-  plot_bar(subset_samples(ps_ra_ice_top), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o_top) +
  ggtitle("Ice") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())

o4_top <-  plot_bar(subset_samples(ps_ra_water_top), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  theme(strip.text=element_text(face="bold")) +
  scale_fill_manual(values=palette_o_top) +
  ggtitle("Water") +
  theme(legend.position = "none") +
  theme(axis.title.x = element_blank(), axis.text.x = element_blank())


#plot order graph with common legend
bottom_row_O_top<- plot_grid(o3_top,o4_top,o2_top, labels = c("B","C","D"), ncol = 3, nrow = 1)


O_grid_top <-plot_grid(o1_top,bottom_row_O_top, labels = c("A",""), ncol = 1)

o_legend_top <- plot_bar(subset_samples(ps_ra_CCC_top), fill="order")+
  geom_bar(aes(fill=order), stat="identity",position="stack") +
  scale_fill_manual(values=palette_o_top) +
  ggtitle("CCC") +
  theme(legend.position = "bottom") 


legendO_top <- get_legend(
  o_legend_top +
    theme(legend.position = "bottom")
)

plot_grid(O_grid_top, legendO_top, ncol = 1, nrow = 2, rel_heights = c(100, 0.1))
  
plot_grid(O_grid_top, legendO_top, ncol = 1, rel_heights = c(100, 0.1)) 
```


```{r} 
#corncob install for differential abundance of ASVs

#devtools::install_github("bryandmartin/corncob")



library(corncob);packageVersion("corncob")

library(phyloseq)
library(magrittr)
#data(soil_phylo)
ps.dc

otu_table(ps.dc)[1:3, 1:3]
tax_table(ps.dc)[1:3, ]

ps_corn<-ps.dc %>%
  phyloseq::subset_samples(name %in% c("E2","E3","E5", "E6","E7","E8","E9","E10","E12","E13","W2","W3","W5"))

ps_corn #13 samples pass through this filter step to compare CCCs to water samples

ps_corn2<-ps.dc%>%
  phyloseq::subset_samples(name %in% c("E2","E3","E5", "E6","E7","E8","E9","E10","E12","E13","I1","W2","W3","W5"))

#skating rink CCCs only
ps_corn3<-ps.dc%>%
  phyloseq::subset_samples(name %in% c("E2","E3","E5", "E6","E7","E8","E9","I1","W2","W3","W5"))



```

```{r}
set.seed(1)
da_analysis <- differentialTest(formula = ~ type,
                                 phi.formula = ~ type,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ type,
                                 test = "Wald", boot = FALSE,
                                 data = ps_corn,
                                 fdr_cutoff = 0.05)
```

```{r}
da_analysis$significant_taxa
plot(da_analysis)


set.seed(1)
da_analysis2 <- differentialTest(formula = ~ type,
                                 phi.formula = ~ type,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ type,
                                 test = "Wald", boot = FALSE,
                                 data = ps_corn2,
                                 fdr_cutoff = 0.05)
                                 

set.seed(1)
da_analysis_b <- differentialTest(formula = ~ type,
                                 phi.formula = ~ type,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ type,
                                 test = "Wald", boot = FALSE,
                                 data = ps_corn2,
                                 B=1000,
                                 fdr_cutoff = 0.05)
str(ps_ra_CCC)
#model never converges, sample size too small to run differential analysis for CCCs by spatial location in WWC                                
# set.seed(1)
# da_analysis_c <- differentialTest(formula = ~ CCC_short,
#                                  phi.formula = ~ CCC_short,
#                                  formula_null = ~ 1,
#                                  phi.formula_null = ~ CCC_short,
#                                  test = "Wald", boot = FALSE,
#                                  data = ps_ra_CCC,
#                                  B=1000,
#                                  fdr_cutoff = 0.05)



#run differential analysis for Skating Rink samples only

set.seed(1)
da_analysis_d <- differentialTest(formula = ~ type,
                                 phi.formula = ~ type,
                                 formula_null = ~ 1,
                                 phi.formula_null = ~ type,
                                 test = "Wald", boot = FALSE,
                                 data = ps_corn3,
                                 B=1000,
                                 fdr_cutoff = 0.05)

da_analysis_d$significant_taxa

```

```{r}

da_analysis2$significant_taxa
fdr_asv <- da_analysis2$significant_taxa
fdr_asv2 <- c("ASV_5", "ASV_14", "ASV_17", "ASV_22", "ASV_23", "ASV_25", "ASV_35","ASV_37", "ASV_41", "ASV_42", "ASV_47","ASV_58", "ASV_87", "ASV_178", "ASV_295")
fdr_asv
plot(da_analysis2)
png(file="WWC_fdr2.png", width=800, height = 600)
plot(da_analysis2)
dev.off()
plot(da_analysis2, level = c("genus", "species"))
plot(da_analysis2, level = c("order","genus", "species"))
png(file="WWC_fdr_ogs.png", width=800, height = 600)
plot(da_analysis2, level = c("order","genus", "species"))

plot(da_analysis_b, level = c("order","genus", "species"))
plot(da_analysis_b, level = c("order","genus", "species"))

print(summary(da_analysis2))
#which(is.na(da_analysis2$p)) %>% names

#pull fasta ASVs for differential taxa
head("/Users/eeggleston/OneDrive - Middlebury College/WWC/ASVsF.fa")


```


```{r}
#function to extract results from corncob differential analysis
sig_models <- function(cc_results){
  df <- data.frame()
  for(i in 1:length(cc_results$significant_taxa)){
    taxa_name <- cc_results$significant_taxa[i]
    taxa_df <- cc_results$significant_models[[i]]$coefficients %>%
      as.data.frame() %>%
      rownames_to_column("covariate") %>%
      mutate(taxa_name = taxa_name)
    df <- bind_rows(df, taxa_df)
  }
  
  df <- clean_names(df) %>%
  select(covariate, estimate, std_error, t_value, fdr = pr_t, taxa_name)
  return(df)
}

#parse data with and without intercept (CCC)
sig.corn.data_noIntercept <-sig_models(da_analysis_b)%>%
  filter(!grepl("Intercept", covariate)) #removes Intercept data for CCCs
palette <- distinctColorPalette(14)
px <- c("#DD5CBD", "#7B76D5", "#6FE195", "#93EA53" ,"#AD44E3" ,"#DE6660", "#D7D3D9" ,"#D3A47B" ,"#D59BCD", "#DCD75E", "#84E2DB", "#6C8C8E", "#D0E1AD" ,"#83B0DF")

barplot(c(1:14), col=palette)

sig.corn.data_all <-sig_models(da_analysis_b) #with CCC intercept

physeq_genus_tt <- tax_table(ps_corn2) %>%
  as.data.frame() %>%
  rownames_to_column("taxa_name")

scdn <- sig.corn.data_noIntercept %>%
  left_join(physeq_genus_tt)

sig.corn.data_all <- sig.corn.data_all %>%
  left_join(physeq_genus_tt)

scdn_filt <- scdn %>%
  filter(covariate== c("mu.typeIce", "mu.typeWater"))%>%
  arrange(estimate)

#filter data and sort levels based on Ice or Water and estimate value
scdn_ice <- scdn_filt %>%
  filter(covariate=="mu.typeIce") %>%
  arrange(desc(estimate))%>%
  mutate_if(is.character, ~replace_na(.,"unclassified")) %>%
  mutate(asvgenus=paste(genus,"(",taxa_name,")"))


scdn_ice.b <- scdn_ice[order(-scdn_ice$estimate),]
scdn_ice.b$taxa_name <-factor(scdn_ice.b$taxa_name, levels = c("ASV_87","ASV_17","ASV_42","ASV_41","ASV_23","ASV_178", "ASV_295","ASV_12", "ASV_5","ASV_14", "ASV_47", "ASV_22","ASV_35","ASV_25"))
scdn_ice.b$asvgenus <-factor(scdn_ice.b$asvgenus, levels = c("Massilia ( ASV_87 )","Listeria ( ASV_17 )", "Chryseobacterium ( ASV_42 )", "Nocardioides ( ASV_41 )",
"Acinetobacter ( ASV_178 )", "Enterococcus ( ASV_23 )","unclassified ( ASV_295 )", 
"Lactobacillus ( ASV_5 )","unclassified ( ASV_14 )",  "unclassified ( ASV_47 )",    
"Euzebya ( ASV_22 )", "Pseudonocardia ( ASV_35 )","unclassified ( ASV_25 )" ))

scdn_ice$asvgenus
scdn_ice.b$asvgenus

scdn_water <- scdn_filt %>%
  filter(covariate=="mu.typeWater") %>%
  arrange(desc(estimate)) %>%
  mutate_if(is.character, ~replace_na(.,"unclassified")) %>%
  mutate(asvgenus=paste(genus,"(",taxa_name,")"))



scdn_water.b <- scdn_water[order(-scdn_ice.b$estimate),]
scdn_water.b$taxa_name <-factor(scdn_water.b$taxa_name, levels = c("ASV_87","ASV_17","ASV_42","ASV_41","ASV_23","ASV_178", "ASV_295","ASV_12", "ASV_5","ASV_14", "ASV_47", "ASV_22","ASV_35","ASV_25"))
scdn_water.b$asvgenus <-factor(scdn_water.b$asvgenus, levels = c("Massilia ( ASV_87 )","Listeria ( ASV_17 )", "Chryseobacterium ( ASV_42 )", "Nocardioides ( ASV_41 )",
"Acinetobacter ( ASV_178 )", "Enterococcus ( ASV_23 )","unclassified ( ASV_295 )", 
"Lactobacillus ( ASV_5 )","unclassified ( ASV_14 )",  "unclassified ( ASV_47 )",    
"Euzebya ( ASV_22 )", "Pseudonocardia ( ASV_35 )","unclassified ( ASV_25 )" ))

scdn_water$asvgenus
scdn_water.b$asvgenus
scdn_water$taxa_name
scdn_water.b$taxa_name

#plot differential analysis
scdn_ice_plot <-  ggplot(scdn_ice.b, aes(x = asvgenus, y = estimate, fill = family)) +
  geom_errorbar(aes(ymin = estimate-std_error, ymax = estimate+std_error), color = "black", width = .3,
                position=position_dodge(.9)) +
  geom_point(size = 4, pch = 21) + 
  ylim(-6,6)+
  scale_x_discrete(limits = rev(levels(scdn_ice.b$asvgenus))) +
  scale_fill_manual(values = px) +
  coord_flip() +
  theme_bw() +
  #theme(legend.position = "none") +
  theme(plot.title = element_text(hjust = 0.5))+
  labs(x = "Differential Taxa", y = "Coefficient", title = "Ice") +
  theme(axis.title.y=element_text(hjust=0.5))+
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgray")
 

scdn_ice_plot

scdn_water_plot <- ggplot(scdn_water.b, aes(x = asvgenus, y = estimate, fill = family)) +
  geom_errorbar(aes(ymin = estimate-std_error, ymax = estimate+std_error), color = "black", width = .3, position=position_dodge(.9)) +
  geom_point(size = 4, pch = 21) + 
  ylim(-6,6)+
  scale_x_discrete(limits = rev(levels(scdn_water.b$asvgenus))) +
  scale_fill_manual(values = palette) +
  coord_flip() +
  theme_bw() +
  theme(legend.position = "none") +
  theme(plot.title = element_text(hjust = 0.5))+
  labs(x = " ", y = "Coefficient", title = "Water") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "darkgray")


legend_df <- get_legend(
  scdn_ice_plot +
    guides(fill=guide_legend(title = NULL))+
    theme(legend.position = "bottom"))

legend_df2 <- get_legend(
  scdn_ice_plot +
    guides(fill=guide_legend(title = NULL))+
    theme(legend.position = "right"))

legend_df2

ice.water_plot <-plot_grid(scdn_ice_plot,scdn_water_plot, labels = c("A","B"), nrow = 1)



plot_grid(ice.water_plot,legend_df, ncol = 1,rel_heights = c(1, .5))

plot_grid(scdn_ice_plot,scdn_water_plot, legend_df2,labels= c("A","B"), ncol = 3, rel_widths = c(1.5,1.5,0.5))


png(file="WWC_DA.png", width=12, height = 8, units='in', res = 300)
plot_grid(scdn_ice_plot,scdn_water_plot, legend_df2,labels= c("A","B"), ncol = 3, rel_widths = c(1.5,1.5,0.5))
dev.off()

```

```{r}
#microbial richness, alpha & beta diversity
#devtools::install_github("adw96/breakaway")
library(breakaway); packageVersion("breakaway") # v.4.8.0

ba <- breakaway(ps.noncontam)
ba
plot(ba, ps.noncontam, color = "type")

ba1 <- breakaway(ps_corn2)
ba1
plot(ba1, ps_corn2, color = "type")


summary(ba1)$estimate
summary(ba1)$error
bF <- betta(summary(ba1)$estimate,
            summary(ba1)$error,
            make_design_matrix(ps_corn2,"type"))
bF2 <- betta(summary(ba1)$estimate,
            summary(ba1)$error,
            make_design_matrix(ps_corn2,"name"))
bF
bF2

bF_pic <- betta_pic(summary(ba1)$estimate,summary(ba1)$error, mymain = "Breakaway alpha diversity estimates",
          mycol = "type")

plot(bF_pic)

###DIVERSITY DIFFERENCES BETWEEN sample type###
bF$table



write.table(bF$table, "betaFsOnly.tsv", sep="\t", quote = F, col.names=NA)





```

